packages/maskbook/src/utils/native-rpc/types.ts

```diff
@@ -0,0 +1,27 @@
+import type { PersonaWithProviders } from '../../database'
+/**
+ * JSON RPC calls that can be called from the native side.
+ * JS = server, native = client
+ */
+export interface WebviewAPIs {
+    web_echo<T>(arg: T): Promise<T>
+    getUrl(path: string): string
+    getStorage(key: string): Promise<browser.storage.StorageObject>
+    getConnectedPersona(): Promise<PersonaWithProviders[]>
+}
+export interface SharedNativeAPIs {}
+/**
+ * JSON RPC calls that can be called if it is running on iOS.
+ * JS = client, iOS = server
+ */
+export interface iOSNativeAPIs extends SharedNativeAPIs {
+    scanQRCode(): Promise<string>
+    log(...args: any[]): Promise<void>
+}
+/**
+ * JSON RPC calls that can be called if it is running on Android.
+ * JS = client, Android = server
+ */
+export interface AndroidNativeAPIs extends SharedNativeAPIs {
+    android_echo(arg: string): Promise<string>
```

Comment:
@jk234ert would you like to add a `scanQRCode(): Promise<string>` method for Android so we can move this API to `SharedNativeAPIs`?

Reply:
Yes, a native QR scan would be much better.

Reply:
If the API is ready please cc me @jk234ert

Reply:
Separate issue created for the Android native scan QR code feature:
#2039 
@Jack-Works this needs some work on the Android native side, as an enhancement.